cmake/CMakeLists.txt

```diff
@@ -1,15 +1,19 @@
 # Minimum CMake required
 cmake_minimum_required(VERSION 2.8.12)
 
-# Project
-project(protobuf C CXX)
+if(protobuf_VERBOSE)
+  message(STATUS "Protocol Buffers Configuring...")
+endif()
 
 # CMake policies
 cmake_policy(SET CMP0022 NEW)
 
+# Project
+project(protobuf C CXX)
+
 # Options
-option(protobuf_VERBOSE "Enable for verbose output" OFF)
```

Comment:
if you check it before it's defined, it evaluates as false, which in this case is OK since it defaults to off, meaning that if a user has set it it will either already be defined from the command line or will be set in the cache.

It's not really the best practice though, you're right.